Assorted File::stat changes and clean-up

mauke opened the pull request:
Building on #23508, here are some bigger changes:

- Make `File::stat::stat` and `File::stat::lstat` always return a scalar, even in list context. In particular, the failure case now always returns `undef`, not the empty list.
- Implement the `CORE::stat` (and `CORE::lstat`) behavior of checking `$_` if no arguments are given.
- Document `stat(\*_)` syntax to reuse previous stat results.
- Add more examples to the documentation.
- Add more links to the documentation.
- Clean up whitespace.

---------------------------------------------------------------------------------
* This set of changes requires a perldelta entry, and it is included.

mauke committed:
File::stat: always return a scalar, even in list context

Previously, a failing stat would return an empty list in list context,
not undef. This can cause surprises in code like:

    foo(stat $file);  # would call foo() without arguments if stat fails

    my %hash = (
        stat => stat($file),
        # Odd number of elements in hash assignment if stat fails
    );

mauke committed:
File::stat: default to $_ if called without arguments

mauke committed:
File::stat: we have signatures, so use them

mauke committed:
File::stat: show examples of (not) overriding stat

That is, show how to access File::stat::stat without overriding anything
and how to access CORE::stat despite overriding stat.

mauke committed:
File::stat: consistently indent verbatim blocks by 4 spaces

Previously, the POD used a mix of 1-space and 4-space indents.

Also, hyperlink stat(2) and Class::Struct.

mauke committed:
File::stat: add equivalent GH IDs to RT ticket references

mauke committed:
File::stat: expand tabs, delete trailing spaces

Most lines in these files used spaces for indentation, but a few of them
still had tabs. Keep things consistent by using spaces everywhere.

Grinnz commented on lib/File/stat.pm:
```diff
@@ -260,13 +262,13 @@ mtime,
-blocks.  
+blocks.
 
 As of version 1.02 (provided with perl 5.12) the object provides C<"-X">
```
Another "as of version" paragraph could be added mentioning the added support of implicit `$_`; probably sufficient to document the context sensitivity fix in perldelta

mauke replied:
Done.

Grinnz commented on lib/File/stat.pm:
```diff
@@ -1,4 +1,4 @@
-package File::stat 1.14;
+package File::stat 1.16;
```
Doesn't need an additional version bump beyond 1.15 unless this ends up in a different release than the other PR

mauke replied:
Unbumped. :-)

mauke committed:
File::stat: document/test stat \*_ syntax to reuse stat buffer

I'm not sure why the "unexported and undocumented populate() function"
had to be mentioned here. The more straightforward `stat(\*_)` syntax
works on all perl versions back to (at least) 5.8.9 and is compatible
with CORE::stat, too.

mauke committed:
perldelta for File::stat 1.15

Grinnz commented on lib/File/stat.pm:
```diff
@@ -286,23 +288,41 @@ variables named with a preceding C<st_> in front their method names.
+    use File::stat;
+    my ($dev, $ino, $mode) = CORE::stat($file);
+
+As of version 1.16 (provided with perl 5.44) C<stat> and C<lstat> can be
```
1.15, now :smiley:
